### PR TITLE
New package: jdx.mise version 2024.12.6

### LINKED ISSUE #3425: Create winget package
In Windows there is a way to install packages like on most linux distros with a package manager. It's called `winget` and is officially supported by Microsoft.

Offering mise to be installed via winget would make the distribution on Windows much easier.

If there is interest in this, I could contribute to it.
Creating a package is pretty straight forward with `winget-create`.

### PR DESCRIPTION
**Pull request by jdx**
###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/winget-pkgs/pull/197444)

Fixes https://github.com/jdx/mise/issues/3425

### CODE REVIEW

**Comment by jdx** [comment]
@microsoft-github-policy-service agree

**Comment by jdx** [comment]
not sure why this is failing, it works locally:

```
winget install --manifest .\manifests\m\mise\jdx\2024.12.4 --ignore-local-archive-malware-scan
```

**Comment by jdx** [comment]
it seems to be failing in "installer validation" but I ran `winget validate` and it passed locally. Unless I'm blind I don't see any actual error coming from CI:

```
12/10/2024 5:32:18 AM

****************************************************************************************
Installation Verification
****************************************************************************************
Status: Started
============================================================================== 
Task         : Installation Validation
Description  : Invoke an Azure Function
Version      : 1.220.0
Author       : Microsoft Corporation
Help URL     : https://docs.microsoft.com/azure/devops/pipelines/tasks/utility/azure-function
============================================================================== 
POST ***?code=***&
Request body: {
  "operationId": "WinGetSvc-Validation-29-197444-20241210-1",
  "buildId": "34420",
  "planUrl": "[https://dev.azure.com/shine-oss/",](https://dev.azure.com/shine-oss/%22,)
  "hubName": "build",
  "projectId": "8b78618a-7973-49d8-9174-4360829d979b",
  "planId": "61c74b16-4f95-450b-a3b6-7233e899dbda",
  "jobId": "68997d86-8ece-5ae8-3ac5-ccdbfa458731",
  "timelineId": "61c74b16-4f95-450b-a3b6-7233e899dbda",
  "taskInstanceId": "b79910ce-dca5-5e27-ad7f-bd285eecb5a2",
  "authToken": ***
  "repository": "microsoft/winget-pkgs",
  "pipelineType": "ValidationPipeline",
  "pullRequestNumber": "197444"
}
				Response Code: OK
				Response: 

12/10/2024 5:36:16 AM

****************************************************************************************
Installation Verification
****************************************************************************************
Status: Completed
Result: Result: Failed
```

**Comment by o-l-a-v** [comment]
When testing the manifest locally like so:

```pwsh
# Path
$ManifestDirPath = [System.IO.Path]::Combine(
    ([System.IO.Directory]::GetParent($psEditor.GetEditorContext().CurrentFile.Path).'FullName'),
    'Mise v2024.12.4'
)

# Try validating
winget validate --manifest $ManifestDirPath

# [admin] Enable required settings
gsudo {
    winget settings --enable LocalArchiveMalwareScanOverride
    winget settings --enable LocalManifestFiles
}

# Try installing
winget install --manifest $ManifestDirPath --installer-type portable --verbose

# Uninstall
winget uninstall --manifest $ManifestDirPath --purge

# [admin] Disable settings that was enabled
gsudo {
    winget settings --disable LocalArchiveMalwareScanOverride
    winget settings --disable LocalManifestFiles
}
```

On following command I get an error saying malware was detected:

```pwsh
PS > winget install --manifest $ManifestDirPath --installer-type portable --verbose

Found mise [mise.jdx] Version 2024.12.4
This application is licensed to you by its owner.
Microsoft is not responsible for, nor does it grant any licenses to, third-party packages.
Downloading https://github.com/jdx/mise/releases/download/v2024.12.4/mise-v2024.12.4-windows-x64.zip
  ██████████████████████████████  9.60 MB / 9.60 MB
Successfully verified installer hash
Archive scan detected malware. To override this check use --ignore-local-archive-malware-scan

PS >
```

Which might be why the CI/CD tests fail? 🤔

**Comment by jdx** [comment]
that's very helpful! I'll give that a go

**Comment by FilipDeVos** [comment]
I'm fairly certain that this is failing because winget does not support zip unpacking portable applications. It wants one of installer types (exe, msi, msix, inno, wix, nullsoft, appx) 

I am aware that there is a reference to "zip" installer type, but I believe it is intended for installers inside zip files.

**Comment by o-l-a-v** [comment]
Here's an example with Azure CLI which also had a portable option with a ZIP:

* <https://github.com/microsoft/winget-pkgs/blob/master/manifests/m/Microsoft/AzureCLI/2.65.0/Microsoft.AzureCLI.installer.yaml>

**Comment by FilipDeVos** [comment]
Download the log from this url: https://dev.azure.com/shine-oss/8b78618a-7973-49d8-9174-4360829d979b/_apis/build/builds/34420/artifacts?artifactName=InstallationVerificationLogs&api-version=7.1&%24format=zip

It is failing due to MS Defender Antivirus finding threats. 

You can submit the file for analysis to https://www.microsoft.com/en-us/wdsi/filesubmission

**Comment by jdx** [comment]
still failing with the updated zip, I'll retry once the antivirus stuff is worked out

**Comment by jdx** [comment]
looks good now!

File: manifests/m/mise/jdx/2024.12.6/mise.jdx.installer.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+PackageIdentifier: mise.jdx
+PackageVersion: 2024.12.6
+InstallerType: zip
+NestedInstallerType: portable
+NestedInstallerFiles:
+- RelativeFilePath: mise/bin/mise.exe
+UpgradeBehavior: install
+ReleaseDate: 2024-12-11
+Installers:
```
Comment (stephengillie) [inline manifests/m/mise/jdx/2024.12.6/mise.jdx.installer.yaml:12]:
```suggestion
Dependencies:
  PackageDependencies:
    - PackageIdentifier: Microsoft.VCRedist.2015+.x64
Installers:
```

**Review by stephengillie — CHANGES_REQUESTED** [review]
This package appears to require `VCRedist`:
image

**Comment by o-l-a-v** [comment]
About the package identifier:

* Isn't the publisher / author here "JDX", not "mise"?
  * Thus, the WinGet package identifier should be `jsx.mise` and placed in "J": <https://github.com/microsoft/winget-pkgs/tree/master/manifests/j>
* What name should be used? Just "Mise" or "Mise en place"?
  * @nagromc made a deal about using the full name over at the Scoop PR: <https://github.com/ScoopInstaller/Main/pull/6374#issuecomment-2528424595>
  * Decide on one name for consistency, and use it across all package managers for consistency.

**Comment by o-l-a-v** [comment]
> I don't think any package manager uses mise-en-place for the name except scoop

Chocolatey currently does.

* <https://community.chocolatey.org/packages/mise>

(I noticed this after I posted my last reply)

**Comment by jdx** [comment]
oh it seems these package managers separate the concept of an "id" and a "name"—I'm mostly familiar with homebrew which does not. I suppose we can use mise-en-place—though I consider these names totally interchangeable

**Comment by stephengillie** [comment]
Verify:
image

Our packages are sorted by `PackageIdentifier` - a composite of `DeveloperName.PackageName`, and generally must follow the developer or publisher's conventions for spelling, punctuation, suffixes, etc. These should match the `Developer` and `PackageName` fields in the locale file, but I'm not currently aware of any logic to check for matching between the fields.

**Review by stephengillie — APPROVED** [review]

**Pull request merged**